src/Database/Models.php

```diff
@@ -204,14 +206,19 @@ public static function isOwnedBy(Model $authority, Model $model)
      * @param  string|\Closure  $attribute
      * @param  \Illuminate\Database\Eloquent\Model  $authority
      * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @param  bool  $isDefaultAttribute
      * @return bool
      */
-    protected static function isOwnedVia($attribute, Model $authority, Model $model)
+    protected static function isOwnedVia($attribute, Model $authority, Model $model, $isDefaultAttribute = false)
     {
         if ($attribute instanceof Closure) {
             return $attribute($model, $authority);
         }
 
+        if ($isDefaultAttribute && !array_key_exists($attribute, $model->getAttributes())) {
```

Comment:
I wish there were a better solution to this. `$model->getAttributes()` is quite expensive.

Maybe we could instead toggle the `preventAccessingMissingAttributes` back and forth before and after checking 🤔

Reply:
That would also be nice.

Anyway, it's something I thought. I think if you want to implement it in some way that you think is best (I mean the entire PR), I'll just close this PR. If this PR served to show this behavior, I consider myself satisfied.

Reply:
The only thing I'm not sure about is the global natore of it. The setting in the `Elqoeunt\Model` class sets a property on `static::`, which I guess means you can override it per modal type. Not sure how switching it back and forth could work 🤔